sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/IVmMetadataProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals.Statsbeat
```

Comment:
Instead of introducing an interface we could introduce a static class like this one https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs

Reply:
static class looks good. how would we replace this behavior in a unit test?

Reply:
By setting the func to whatever test needs to be.

Reply:
a static func will have side effects for other tests. this is something i specifically want to avoid.

Reply:
we could make that class non-static too.